modules/c++/sys/source/OSUnix.cpp

```diff
@@ -346,6 +346,53 @@ void sys::OSUnix::getMemInfo(size_t &totalPhysMem, size_t &freePhysMem) const
 #endif
 }
 
+std::string sys::OSUnix::getCurrentExecutable(
+        const std::string& argvPathname) const
+{
+    std::vector<std::string> possibleSymlinks;
+
+    // Linux
+    possibleSymlinks.push_back(sys::Path::joinPaths(
+            sys::Path::delimiter()[0] + std::string("proc"),
+            sys::Path::joinPaths("self", "exe")));
+
+    // Solaris
+    possibleSymlinks.push_back(sys::Path::joinPaths(
+            sys::Path::delimiter()[0] + std::string("proc"),
+            sys::Path::joinPaths("self",
+            sys::Path::joinPaths("path", "a.out"))));
+
+    for (size_t ii = 0; ii < possibleSymlinks.size(); ++ii)
+    {
+        struct stat linkStatus;
```

Comment:
It does get confused without struct. Everything else is fixed.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Thursday, July 14, 2016 8:49 PM
To: mdaus/coda-oss <coda-oss@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [mdaus/coda-oss] Current path (#146)


In modules/c++/sys/source/OSUnix.cpp<https://github.com/mdaus/coda-oss/pull/146#discussion_r70907654>:

> +    std::vector<std::string> possibleSymlinks;

> +

> +    // Linux

> +    possibleSymlinks.push_back(sys::Path::joinPaths(

> +            sys::Path::delimiter()[0] + std::string("proc"),

> +            sys::Path::joinPaths("self", "exe")));

> +

> +    // Solaris

> +    possibleSymlinks.push_back(sys::Path::joinPaths(

> +            sys::Path::delimiter()[0] + std::string("proc"),

> +            sys::Path::joinPaths("self",

> +            sys::Path::joinPaths("path", "a.out"))));

> +

> +    for (size_t ii = 0; ii < possibleSymlinks.size(); ++ii)

> +    {

> +        struct stat linkStatus;

I don't think you need the struct keyword here (unless the compiler was getting confused with the stat() function)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/mdaus/coda-oss/pull/146/files/4c424c4dcac4c8689dcd553518eb6da388b0cbb7#r70907654>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFRAYIp6pV0b6nCDHqncQEmCUfCzY3l0ks5qVtkWgaJpZM4JLdyF>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.